declarations_site/spotter/utils.py

```diff
@@ -255,3 +267,49 @@ def create_notify(task, report):
     send_found_notify(notify)
     notify.save()
     return notify
+
+
+def save_search_task(user, query, deepsearch=True, query_params='', chat_data=''):
+    if len(query) < 2:
+        raise ValueError('Не вдалось створити завдання з пустим запитом.')
+
+    if len(query) > 100 or len(query.split(' ')) > 10:
+        raise ValueError('Не вдалось створити завдання з таким довгим запитом.')
+
+    if not user.email:
+        raise ValueError('Не вдалось створити завдання без адреси електронної пошти.')
+
+    # don't add twice
+    if SearchTask.objects.filter(user=user, query=query,
+            deepsearch=deepsearch, query_params=query_params, is_deleted=False).exists():
+        raise ValueError('Таке завдання вже існує.')
+
+    task = SearchTask(user=user, query=query, deepsearch=deepsearch)
+    task.query_params = query_params
+    task.chat_data = chat_data
+    task.save()
+
+    if not first_run(task):
```

Comment:
Again, ideally you should raise an exception in first_run and here you should catch it (rather rely on implicit implementation details).